packages/stores/src/queries/pool-incentives/incentivized-pools.ts

```diff
@@ -119,51 +121,46 @@ export class ObservableQueryIncentivizedPools extends ObservableChainQuery<Incen
   );
 
   /**
-   * Computes external incentive APY for the given duration
+   * Computes the external incentive APR for the given gaugeId and denom
    */
-  readonly computeExternalIncentiveAPYForSpecificDuration = computedFn(
+  readonly computeExternalIncentiveGaugeAPR = computedFn(
     (
       poolId: string,
-      duration: Duration,
+      gaugeId: string,
+      denom: string,
       priceStore: IPriceStore,
-      fiatCurrency: FiatCurrency,
-      allowedGauges: ExternalGauge[]
+      fiatCurrency: FiatCurrency
     ): RatePretty => {
-      if (!allowedGauges.length) {
+      const observableGauge = this.queryGauge.get(gaugeId);
+
+      if (observableGauge.remainingEpoch < 1) {
         return new RatePretty(new Dec(0));
       }
 
-      const externalGauge = allowedGauges.find((externalGauge) => {
-        return (
-          duration.asMilliseconds() === externalGauge.duration.asMilliseconds()
-        );
-      });
+      const chainInfo = this.chainGetter.getChain(this.chainId);
+
+      const mintCurrency = chainInfo.findCurrency(denom);
 
-      if (!externalGauge?.rewardAmount) {
+      if (!mintCurrency?.coinGeckoId) {
```

Comment:
Is it right that external incentive should have a price info?

Reply:
Yeah I think it should have frontend configuration.

Check out this issue:
https://github.com/osmosis-labs/osmosis-frontend/issues/853